Mathlib/SetTheory/ZFC/Basic.lean

```diff
@@ -892,6 +892,24 @@ theorem singleton_nonempty (u : ZFSet) : ZFSet.Nonempty {u} :=
 theorem mem_pair {x y z : ZFSet.{u}} : x ∈ ({y, z} : ZFSet) ↔ x = y ∨ x = z := by
   simp
 
+@[simp]
+theorem pair_self (x : ZFSet) : {x, x} = ({x} : ZFSet) := by
```

Comment:
The `Set` and `Finset` versions are called `pair_eq_singleton`; I suggest using the same name here, and using `pair_eq_singleton_iff` for your lemma below.